debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -234,6 +234,17 @@ protected void recoverRecords(Tables schema, DdlParser ddlParser, Consumer<Histo
         }
     }
 
+    private long getEndOffsetOfDbHistoryTopic(KafkaConsumer<String, String> historyConsumer) {
+        Map<TopicPartition, Long> offsets = historyConsumer.endOffsets(Collections.singleton(new TopicPartition(topicName, PARTITION)));
```

Comment:
I couldn't make it work via `historyConsumer.assigments()` which would return an empty set (I suppose assignment hasn't happened yet). Hence I'm creating the `TopicPartition` object myself.

Reply:
I think that's fine.